File: python/1572.Matrix-Diagonal-Sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        left = 0
        right = n - 1
        total = 0

        while left < n:
            if left == right:
                total += mat[left][left]
            else:
                total += mat[left][left] + mat[left][right]
            
            left += 1
            right -= 1

        return total

File: javascript/1572.Matrix-Diagonal-Sum.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function (mat) {
  const rows = mat.length;
  const cols = mat[0].length;

  let leftDiagIdx = 0;
  let rightDiagIdx = rows - 1;
  let sum = 0;

  while (leftDiagIdx < rows) {
    if (leftDiagIdx === rightDiagIdx) {
      sum += mat[leftDiagIdx][leftDiagIdx];
    } else {
      sum += mat[leftDiagIdx][leftDiagIdx] + mat[leftDiagIdx][rightDiagIdx];
    }

    leftDiagIdx += 1;
    rightDiagIdx -= 1;
  }

  return sum;
};
